static/styles/night_mode.css

```diff
@@ -50,25 +30,14 @@ body.night-mode {
     }
 
     .compose-send-status-close {
-        color: hsl(0, 0%, 100%);
+        color: var(--color-text);
         opacity: 1;
     }
 
     .compose-send-status-close:hover {
         opacity: 0.4;
     }
 
-    .message_embed .data-container::after {
-        background: linear-gradient(0deg, hsl(212, 28%, 18%), transparent 100%);
-    }
```

Comment:
I think the end result here looks pretty reasonable. The way I'd want to write commits for this sort of migration is to have smaller commits migrating individual types of components, e.g. this line feels complicated enough that it'd be nice to have it in the in small commit that changes this and adjusts `zulip.css` appropriately.